#! /bin/sh /usr/share/dpatch/dpatch-run
## 005_libav-api.dpatch by Colin Watson <cjwatson@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Port to the current libav API.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' aqualung-0.9~beta11~/src/decoder/dec_lavc.c aqualung-0.9~beta11/src/decoder/dec_lavc.c
--- aqualung-0.9~beta11~/src/decoder/dec_lavc.c	2009-08-20 19:11:11.000000000 +0100
+++ aqualung-0.9~beta11/src/decoder/dec_lavc.c	2011-08-18 18:24:13.000000000 +0100
@@ -53,7 +53,7 @@
 
 	if (packet.stream_index == pd->audioStream) {
 
-		avcodec_decode_audio2(pd->avCodecCtx, samples, &n_bytes, packet.data, packet.size);
+		avcodec_decode_audio3(pd->avCodecCtx, samples, &n_bytes, &packet);
 		if (n_bytes > 0) {
 			int i;
 			for (i = 0; i < n_bytes/2; i++) {
@@ -111,7 +111,7 @@
 	file_decoder_t * fdec = dec->fdec;
 	int i;
 
-	if (av_open_input_file(&pd->avFormatCtx, filename, NULL, 0, NULL) != 0)
+	if (avformat_open_input(&pd->avFormatCtx, filename, NULL, NULL) != 0)
 		return DECODER_OPEN_BADLIB;
 
 	if (av_find_stream_info(pd->avFormatCtx) < 0)
@@ -124,7 +124,7 @@
 
 	pd->audioStream = -1;
 	for (i = 0; i < pd->avFormatCtx->nb_streams; i++) {
-		if (pd->avFormatCtx->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO) {
+		if (pd->avFormatCtx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO) {
 			pd->audioStream = i;
 			break;
 		}
